### PR TITLE
Angular: Option to disable angular support and isolate angular dependencies 

### PR DESCRIPTION
**Pull request by torkelo**
* [x] Adds new option `angular_support_enabled` under `[security]`
* [x] Add docs for setting 
* [x] Load angular panels lazily (So they don't cause angular to be bundled in main bundle)
* [x] Load angular app via isolated webpack chunk
* [x] Load angular plugin dependencies only when setting is enabled (exposeToPlugin coreModule etc) 
* [x] Load old legacy AlertTab lazily in it's own chunk (to isolate it from the main bundle) 

This isolates all angular dependencies, and the main angular support is isolated into a single webpack chunk. When angular_support_enabled is set to false no angular dependencies or the main  angular library is loaded. 

Optimized build: 7.13mb -> 6.3mb  (for main entry point) 

This PR does not try to add a load-on-demand feature, which is much harder, especially for external plugins.

### CODE REVIEW

File: conf/defaults.ini [hunk 1]
```diff
@@ -292,6 +292,9 @@ content_security_policy = false
 # $ROOT_PATH is server.root_url without the protocol.
 content_security_policy_template = """script-src 'self' 'unsafe-eval' 'unsafe-inline' 'strict-dynamic' $NONCE;object-src 'none';font-src 'self';style-src 'self' 'unsafe-inline' blob:;img-src * data:;base-uri 'self';connect-src 'self' grafana.com ws://$ROOT_PATH wss://$ROOT_PATH;manifest-src 'self';media-src 'none';form-action 'self';"""
 
+# Controls if old angular plugins are supported or not
```
Comment (ryantxu) [inline conf/defaults.ini:295]: Good place for a comment about the default will change in grafana 9

File: public/app/features/plugins/built_in_plugins.ts [hunk 2]
```diff
@@ -73,6 +70,12 @@ import * as alertGroupsPanel from 'app/plugins/panel/alertGroups/module';
 const geomapPanel = async () => await import(/* webpackChunkName: "geomapPanel" */ 'app/plugins/panel/geomap/module');
 const canvasPanel = async () => await import(/* webpackChunkName: "canvasPanel" */ 'app/plugins/panel/canvas/module');
 const iconPanel = async () => await import(/* webpackChunkName: "iconPanel" */ 'app/plugins/panel/icon/module');
+const graphPanel = async () =>
+  await import(/* webpackChunkName: "alertmanagerPlugin" */ 'app/plugins/panel/graph/module');
+const heatmapPanel = async () =>
+  await import(/* webpackChunkName: "alertmanagerPlugin" */ 'app/plugins/panel/heatmap/module');
+const tableOldPanel = async () =>
+  await import(/* webpackChunkName: "alertmanagerPlugin" */ 'app/plugins/panel/table-old/module');
```
Comment (ryantxu) [inline public/app/features/plugins/built_in_plugins.ts:78]:
```suggestion
const graphPanel = async () =>
  await import(/* webpackChunkName: "graphPlugin" */ 'app/plugins/panel/graph/module');
const heatmapPanel = async () =>
  await import(/* webpackChunkName: "heatmapPlugin" */ 'app/plugins/panel/heatmap/module');
const tableOldPanel = async () =>
  await import(/* webpackChunkName: "tableOldPlugin" */ 'app/plugins/panel/table-old/module');
```

File: conf/defaults.ini (hunk 1 above)
Comment (torkelo) [reply conf/defaults.ini:295]: yes!  :+1:

File: conf/sample.ini [hunk 3]
```diff
@@ -292,6 +292,9 @@
 # $ROOT_PATH is server.root_url without the protocol.
 ;content_security_policy_template = """script-src 'self' 'unsafe-eval' 'unsafe-inline' 'strict-dynamic' $NONCE;object-src 'none';font-src 'self';style-src 'self' 'unsafe-inline' blob:;img-src * data:;base-uri 'self';connect-src 'self' grafana.com ws://$ROOT_PATH wss://$ROOT_PATH;manifest-src 'self';media-src 'none';form-action 'self';"""
 
+# Controls if old angular plugins are supported or not. This will be disabled by default in Grafana v9.
+angular_support_enabled = true
```
Comment (jackw) [inline conf/sample.ini:296]:
```suggestion
;angular_support_enabled = true
```
Based on the rest of `sample.ini` I think this needs to be commented out.

**Review by mckn — APPROVED** [review]
LGTM!

File: conf/sample.ini (hunk 3 above)
Comment (torkelo) [reply conf/sample.ini:296]: ah, good catch!

File: public/app/features/alerting/AlertTab.tsx [hunk 4]
```diff
@@ -57,8 +54,14 @@ class UnConnectedAlertTab extends PureComponent<Props, State> {
...
   };
 
-  componentDidMount() {
-    this.loadAlertTab();
+  async componentDidMount() {
+    if (config.angularSupportEnabled) {
+      await import(/* webpackChunkName: "AlertTabCtrl" */ 'app/features/alerting/AlertTabCtrl');
+      this.loadAlertTab();
+    } else {
+      // TODO probably need to migrate AlertTab to react this
```
Comment (leventebalogh) [inline public/app/features/alerting/AlertTab.tsx:62]: **Nit:** typo?
```suggestion
      // TODO probably need to migrate AlertTab to react
```

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

**Pull request merged**